src/modules/LimitOrders.ts

```diff
@@ -0,0 +1,337 @@
+import {
+  address,
+  ContractCallOptions,
+  ContractConstantCallOptions,
+  Integer,
+  LimitOrder,
+  LimitOrderStatus,
+} from '../../src/types';
+
+const EIP712_ORDER_STRUCT = [
+  { type: 'uint256', name: 'makerMarket' },
+  { type: 'uint256', name: 'takerMarket' },
+  { type: 'uint256', name: 'makerAmount' },
+  { type: 'uint256', name: 'takerAmount' },
+  { type: 'address', name: 'makerAccountOwner' },
+  { type: 'uint256', name: 'makerAccountNumber' },
+  { type: 'address', name: 'takerAccountOwner' },
+  { type: 'uint256', name: 'takerAccountNumber' },
+  { type: 'uint256', name: 'expiration' },
+  { type: 'uint256', name: 'salt' },
+];
+
+const EIP712_ORDER_STRUCT_STRING =
+  'LimitOrder(' +
+  'uint256 makerMarket,' +
+  'uint256 takerMarket,' +
+  'uint256 makerAmount,' +
+  'uint256 takerAmount,' +
+  'address makerAccountOwner,' +
+  'uint256 makerAccountNumber,' +
+  'address takerAccountOwner,' +
+  'uint256 takerAccountNumber,' +
+  'uint256 expiration,' +
+  'uint256 salt' +
+  ')';
+
+export class LimitOrders {
+  private contracts: Contracts;
+  private web3: Web3;
+  private networkId: number;
+
+  // ============ Constructor ============
+
+  constructor(
+    contracts: Contracts,
+    web3: Web3,
+    networkId: number,
+  ) {
+    this.contracts = contracts;
+    this.web3 = web3;
+    this.networkId = networkId;
+  }
+
+  // ============ On-Chain Approve / On-Chain Cancel ============
+
+  public async approveOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.approveOrder(orderHash),
+      options,
+    );
+  }
+
+  public async cancelOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.cancelOrder(orderHash),
+      options,
+    );
+  }
+
+  // ============ Getter Contract Methods ============
+
+  /**
+   * Gets the status and the current filled amount (in makerAmount) of the order from the chain.
+   */
+  public async getOrderState(
+    order: LimitOrder,
+    options?: ContractConstantCallOptions,
+  ): Promise<{status: LimitOrderStatus, makerFilledAmount: Integer}> {
+    const state = await this.contracts.callConstantContractFunction(
+      this.contracts.limitOrders.methods.getOrderState(
+        this.getOrderHash(order),
+        order.makerAccountOwner,
+      ),
+      options,
+    );
+    return {
+      status: parseInt(state[0], 10),
+      makerFilledAmount: new BigNumber(state[1]),
+    };
+  }
+
+  // ============ Signing Methods ============
+
+  /**
+   * Sends order to current provider for signing. Uses the newer 'eth_signTypedData_v3' rpc call
+   * which is compatible with metamask but may not be compatible with all rpc providers.
```

Comment:
can you check that at least one of these is supported by coinbase wallet as well?